arch/risc-v/src/mpfs/mpfs_ethernet.c

```diff
@@ -0,0 +1,3823 @@
+#endif
+
+/* Number of buffers for TX */
+
+#ifndef CONFIG_MPFS_ETHMAC_NTXBUFFERS
+#  define CONFIG_MPFS_ETHMAC_NTXBUFFERS  (8)
+#endif
+
+/* GMAC buffer sizes
+ * REVISIT: do we want to make GMAC_RX_UNITSIZE configurable?
+ * issue when using the MTU size receive block
+ */
+
+#define GMAC_RX_UNITSIZE  (512)                  /* Fixed size for RX buffer */
+#define GMAC_TX_UNITSIZE  CONFIG_NET_ETH_PKTSIZE /* MAX size for Ethernet packet */
+
+/* The MAC can support frame lengths up to 1536 bytes */
+
+#define GMAC_MAX_FRAMELEN (1536)
+#if CONFIG_NET_ETH_PKTSIZE > GMAC_MAX_FRAMELEN
+#  error CONFIG_NET_ETH_PKTSIZE is too large
+#endif
+
+/* for DMA dma_rxbuf_size */
+
+#define MPFS_MAC_RX_BUF_VALUE ((GMAC_RX_UNITSIZE + 63) / 64)
+
+/* We need at least one more free buffer than transmit buffers */
+
+#define MPFS_GMAC_NFREEBUFFERS (CONFIG_MPFS_ETHMAC_NTXBUFFERS + 1)
+
+#ifdef CONFIG_NET_DUMPPACKET
+#  define mpfs_dumppacket(m,a,n) lib_dumpbuffer(m,a,n)
+#else
+#  define mpfs_dumppacket(m,a,n)
+#endif
+
+#ifdef CONFIG_MPFS_HAVE_CORERMII
+#  define CORE_RMII_RESET       (1 << 15)
+#  define CORE_RMII_LOOPBACK    (1 << 2)
+#  define CORE_RMII_FULL_DUPLEX (1 << 1)
+#  define CORE_RMII_100MBIT     (1 << 0)
+#endif
+
+/* Timing *******************************************************************/
+
+/* TX poll delay = 1 seconds.
+ * CLK_TCK is the number of clock ticks per second
+ */
+
+#define MPFS_WDDELAY     (1 * CLK_TCK)
+
+/* TX timeout = 1 minute */
+
+#define MPFS_TXTIMEOUT   (60 * CLK_TCK)
+
+/* PHY reset tim in loop counts */
+
+#define PHY_RESET_WAIT_COUNT (10)
+
+/* PHY re-try coount in loop counts */
+
+#define PHY_RETRY_MAX   300000
+
+/* This is a helper pointer for accessing the contents of the GMAC header */
+
+#define BUF ((struct eth_hdr_s *)priv->dev.d_buf)
+
+#define upper_32_bits(n) ((uint32_t)(((n) >> 16) >> 16))
+#define lower_32_bits(n) ((uint32_t)(n))
+
+/* Interrupt flags */
+
+#define INT_RX     (GEM_INT_RECEIVE_COMPLETE)
+#define INT_TX     (GEM_INT_TRANSMIT_COMPLETE)
+#define INT_ERRORS (GEM_INT_TRANSMIT_BUFFER_UNDERRUN | \
+                    GEM_INT_RECEIVE_OVERRUN | \
+                    GEM_INT_AMBA_ERROR | GEM_INT_RESP_NOT_OK)
+#define INT_ALL    (INT_RX | INT_TX | INT_ERRORS)
+
+/****************************************************************************
+ * Private Types
+ ****************************************************************************/
+
+struct gmac_rxdesc_s
+{
+    uint32_t addr;    /* Buffer address */
+    uint32_t status;  /* RX status and controls */
+#if defined(CONFIG_MPFS_ETHMAC_64BIT_ADDRESS_MODE)
+    uint32_t addr_hi; /* Buffer address high 32-bits */
+    uint32_t unused;
+#endif
+};
+
+/* Transmit buffer descriptor */
+
+struct gmac_txdesc_s
+{
+    uint32_t addr;    /* Buffer address */
+    uint32_t status;  /* TX status and controls */
+#if defined(CONFIG_MPFS_ETHMAC_64BIT_ADDRESS_MODE)
+    uint32_t addr_hi; /* Buffer address high 32-bits */
+    uint32_t unused;  /*  */
```

Comment:
2 spaces indentation.

Reply:
Oh... now i got what you mean, sorry about that.
done.